lib/plugin/opsgenie/types.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2015-2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package opsgenie
+
+// Alert represents an Opsgenie alert
+type Alert struct {
+	ID       string  `json:"id"`
+	Title    string  `json:"title"`
+	Status   string  `json:"status"`
+	AlertKey string  `json:"alert_key"`
+	Body     Details `json:"body"`
+}
+
+// AlertBody represents and Opsgenie alert body
+type AlertBody struct {
+	Message     string      `json:"message,omitempty"`
+	Alias       string      `json:"alias,omitempty"`
+	Description string      `json:"description,omitempty"`
+	Responders  []Responder `json:"responders,omitempty"`
+	Priority    string      `json:"priority,omitempty"`
```

Comment:
Can you please add godoc to all the keys?
Most of the time is obvious but I guess we still recomend to add them to all the fields.
However, others are not that obvious. Eg Alias
I had to see the docs to better understand its goal
https://docs.opsgenie.com/docs/alert-api#create-alert

The same for the `responders.type`